model/project_ref.go

```diff
@@ -313,8 +313,12 @@ func (p *ProjectRef) IsEnabled() bool {
 	return utility.FromBoolPtr(p.Enabled)
 }
 
+// IsPrivate checks if the project ref should be accessed by non-logged in users.
+// If PartialRouteAuthDisabled is set, all project routes require users to be logged in
+// so this function will return false.
 func (p *ProjectRef) IsPrivate() bool {
-	return utility.FromBoolPtr(p.Private)
+	flags, err := evergreen.GetServiceFlags()
+	return err == nil && !flags.PartialRouteAuthDisabled && utility.FromBoolPtr(p.Private)
```

Comment:
This is pedantic but if err != ni, we should just use utility.FromBoolPtr(p.Private) I. think, so we don't return false for actually private projects